aesd-char-driver/main.c

```diff
@@ -307,14 +317,126 @@ loff_t aesd_llseek(struct file *filp, loff_t offset, int whence)
 
 /* update actual f_pos with new calculated value */
     filp->f_pos = new_f_pos;
-    PDEBUG("\nNew f_pos = %lld", filp->f_pos);
+    PDEBUG("\nUpdated f_pos = %lld", filp->f_pos);
 
 exit:
     mutex_unlock(&dev->mutex);
 
     return new_f_pos;
 }
 
+/**
+ *  @name   aesd_adjust_file_offset
+ *  @brief  adjust *f_pos to the desired entry start offset + desired offset within that entry
```

Comment:
@mash9756 I'd add a note that the dev lock must be held by caller for the device corresponding to `filp`.  This isn't obvious unless reviewing the code.
Think about a way to make this more obvious, for instance by passing in a reference to the device buffer.